lib/matplotlib/rcsetup.py

```diff
@@ -60,8 +62,28 @@ def __call__(self, s):
                          % (self.key, s, list(six.itervalues(self.valid))))
 
 
+def _listify_validator(scalar_validator):
+    def f(s):
+        if isinstance(s, six.string_types):
+            try:
+                return [scalar_validator(v.strip()) for v in s.split(',')
+                        if v.strip()]
+            except Exception:
+                # Sometimes, a list of colors might be a single string
+                # of single-letter colornames. So give that a shot.
+                return [scalar_validator(v.strip()) for v in s if v.strip()]
+        elif type(s) in (list, tuple):
+            return [scalar_validator(v) for v in s if v]
+        else:
+            msg = "'s' must be of type [ string | list | tuple ]"
+            raise ValueError(msg)
+    f.__doc__ = scalar_validator.__doc__
+    return f
+
```

Comment:
Very similar to what I had done [here](https://github.com/matplotlib/matplotlib/pull/4810/files#diff-546df25a362e568e615b2a0558df2cbe), but why make it private, the idea I had before was to make this a normal validator, a list validator, with a default function of ``lambda s: s``